### PR TITLE
[BICGSTAB] Add a frequency option for the verbose mode

### PR DESCRIPTION
**Pull request by amontoison**
I will update the other solvers if you're fine with this syntax.
https://ci.appveyor.com/project/dpo/krylov-jl/builds/36383369/job/he4b75f2pcfveayk#L80

### CODE REVIEW

**Comment by dpo** [comment]
Just `frequency` is enough. Or `verbose` could be an int.

**Comment by amontoison** [comment]
I like the second idea.
`verbose=0` -> no display
`verbose=p` with `p>=1` -> display at each multiple of p iterations

**Comment by dpo** [comment]
Ok. Better say `verbose ≤ 0` -> no display.

**Comment by amontoison** [comment]
I updated the PR.

File: src/bicgstab.jl [hunk 1]
```diff
@@ -29,18 +29,20 @@ If BICGSTAB stagnates, we recommend DQGMRES and BiLQ as alternative methods for
 BICGSTAB stops when `itmax` iterations are reached or when `‖rₖ‖ ≤ atol + ‖b‖ * rtol`.
 `atol` is an absolute tolerance and `rtol` is a relative tolerance.
 
-Additional details can be displayed if the `verbose` mode is enabled.
+Additional details can be displayed if the verbose mode is enabled (verbose > 0).
```
Comment (dpo) [inline src/bicgstab.jl:32]:
```suggestion
Additional details can be displayed if verbose mode is enabled (verbose > 0).
```

File: src/bicgstab.jl [hunk 2]
```diff
@@ -29,18 +29,20 @@ If BICGSTAB stagnates, we recommend DQGMRES and BiLQ as alternative methods for
...
+The frequency of the display can be tuned, it is at each multiple of `verbose` iterations.
```
Comment (dpo) [inline src/bicgstab.jl:33]:
```suggestion
Information will be displayed every `verbose` iterations.
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by dpo** [comment]
Thank you.

**Pull request merged**